### PR TITLE
Methods lesson: Provide descriptive link text

### PR DESCRIPTION
**Pull request by CJKbarts**
## Issue

Related to #28290

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [ ] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: ruby/basic_ruby/methods.md [hunk 1]
```diff
@@ -17,18 +18,18 @@ In this lesson, we're going to deconstruct what methods are, examine their behav
...
+One of Ruby's great advantages for new programmers is its large number of built-in methods. You've been using many of them already, probably without even realizing it. Over the course of your learning so far, you have modified strings and other objects in various ways. For example, the [#times loop](https://docs.ruby-lang.org/en/3.3/Integer.html#method-i-times) and [#upto loop](https://docs.ruby-lang.org/en/3.3/Integer.html#method-i-upto) that you learned about in the Loops lesson are both methods that are included as part of Ruby's [`Integer`](https://docs.ruby-lang.org/en/3.3/Integer.html) class.
```
Comment (MaoShizhong) [inline ruby/basic_ruby/methods.md:30]: Missed this one
```suggestion
One of Ruby's great advantages for new programmers is its large number of built-in methods. You've been using many of them already, probably without even realizing it. Over the course of your learning so far, you have modified strings and other objects in various ways. For example, the [#times loop](https://docs.ruby-lang.org/en/3.3/Integer.html#method-i-times) and [#upto loop](https://docs.ruby-lang.org/en/3.3/Integer.html#method-i-upto) that you learned about in the Loops lesson are both methods that are included as part of Ruby's [Integer class](https://docs.ruby-lang.org/en/3.3/Integer.html).
```

File: ruby/basic_ruby/methods.md [hunk 2]
```diff
@@ -60,26 +61,26 @@ puts my_name    #=> "Joe Smith"
...
+It's good practice to start the method name with a lower-case character, because [names that start with capital letters are constants in Ruby](https://stackoverflow.com/questions/10542354/what-are-the-restrictions-for-method-names-in-ruby).
 
 Here are some things you are not allowed to do with your method names:
 
-* <span id="reserved-words">You cannot name your method one of Ruby's approximately 40 reserved words, such as `end`, `while`, or `for`. Checkout the full list [here](http://www.java2s.com/Code/Ruby/Language-Basics/Rubysreservedwords.htm).</span>
-* You cannot use any symbols other than `_`, `?`, `!`, and `=`.
-* You cannot use `?`, `!`, or `=` anywhere other than at the end of the name.
-* You cannot begin a method name with a number.
+- <span id="reserved-words">You cannot name your method one of [Ruby’s approximately 40 reserved words](https://stackoverflow.com/questions/10542354/what-are-the-restrictions-for-method-names-in-ruby), such as `end`, `while`, or `for`.</span>
```
Comment (MaoShizhong) [inline ruby/basic_ruby/methods.md:80]: The href for this link was changed in a previous PR that I missed, so the spreadsheet wasn't updated - oops! However, checking the new link, it seems broken and doesn't lead to the right list. Therefore, the below now contains the doc page with the keywords which I think makes more sense than an SO post or other page anyway.
```suggestion
- <span id="reserved-words">You cannot name your method one of [Ruby’s approximately 40 reserved words](https://docs.ruby-lang.org/en/3.3/keywords_rdoc.html), such as `end`, `while`, or `for`.</span>
```

File: ruby/basic_ruby/methods.md [hunk 3]
```diff
@@ -297,33 +298,36 @@ puts whisper #=> "hello everybody"
...
-2. To get a different take, read the part of the Objects section that discusses [Methods](http://ruby-for-beginners.rubymonstas.org/objects/methods.html) from Ruby Monsta's *Ruby for Beginners*.
-3. For more depth on how you can write your own methods, read the section on [Writing Methods](http://ruby-for-beginners.rubymonstas.org/writing_methods.html) from Ruby Monsta's *Ruby for Beginners*.
-4. Complete the [method](https://github.com/TheOdinProject/ruby-exercises/tree/main/ruby_basics) exercises from the [ruby-exercises repo](https://github.com/TheOdinProject/ruby-exercises) that you previously cloned.
+1. To get a different take, read the part of the Objects section that discusses [Methods](http://ruby-for-beginners.rubymonstas.org/objects/methods.html) from Ruby Monsta's *Ruby for Beginners*.
+1. For more depth on how you can write your own methods, read the section on [Writing Methods](http://ruby-for-beginners.rubymonstas.org/writing_methods.html) from Ruby Monsta's *Ruby for Beginners*.
+1. Complete the Complete the method exercises from the [ruby-exercises repo](https://github.com/TheOdinProject/ruby-exercises) that you previously cloned.
```
Comment (MaoShizhong) [inline ruby/basic_ruby/methods.md:307]:
```suggestion
1. Complete the method exercises from the [ruby-exercises repo](https://github.com/TheOdinProject/ruby-exercises) that you previously cloned.
```

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
Thanks for doing these and also fixing up the lint errors. Just a couple of small changes below then we should be good to merge!

**Comment by CJKbarts** [comment]
I think I got everything. I had some problems trying to squash the new commit into the old one but I think I got it.
Thank you for your feedback

**Review by MaoShizhong — APPROVED** [review]
:rocket: Congrats on your first PR here

**Comment by CJKbarts** [comment]
Thank you

**Comment by CJKbarts** [comment]
Do I delete the branch then?

**Comment by MaoShizhong** [comment]
> Do I delete the branch then?

You can safely delete the branch now that it's been merged.

**Pull request merged**